builder/openstack_id3/step_allocate_ip.go

```diff
@@ -0,0 +1,67 @@
+package openstack_id3
+
+import (
+	"fmt"
+	"github.com/mitchellh/multistep"
+	"github.com/mitchellh/packer/packer"
+
+	"github.com/rackspace/gophercloud"
+	"github.com/rackspace/gophercloud/openstack/compute/v2/extensions/floatingip"
+	"github.com/rackspace/gophercloud/openstack/compute/v2/servers"
+)
+
+type StepAllocateIp struct {
+	FloatingIpPool string
+	FloatingIp     string
+}
+
+func (s *StepAllocateIp) Run(state multistep.StateBag) multistep.StepAction {
+
+	ui := state.Get("ui").(packer.Ui)
+	computeClient := state.Get("compute_client").(*gophercloud.ServiceClient)
+	server := state.Get("server").(*servers.Server)
+
+	var instanceIp *floatingip.FloatingIP
+	if s.FloatingIp != "" {
+		instanceIp.IP = s.FloatingIp
+	} else if s.FloatingIpPool != "" {
+		// Obtain new floating IP. Supports both networks pool id and name
+		newIp, err := floatingip.Create(computeClient, floatingip.CreateOpts{Pool: s.FloatingIpPool}).Extract()
+		if err != nil {
+			err := fmt.Errorf("Error creating floating ip from pool '%s'", s.FloatingIpPool)
+			state.Put("error", err)
+			ui.Error(err.Error())
+			return multistep.ActionHalt
+		}
+		instanceIp = newIp
+		ui.Say(fmt.Sprintf("Created temporary floating IP %s...", instanceIp.IP))
+	}
+	if instanceIp.IP != "" {
```

Comment:
This fails with a nil pointer dereference when not using floating IPs. I was able to fix this by making instanceIP a floatingip.FloatingIP (as opposed to a pointer), like the old builder did.